users/views.py

```diff
@@ -110,47 +113,26 @@ def from_valid(self, form: CustomUserCreationForm) -> HttpResponseRedirect:
         return redirect(self.get_success_url())
 
 
-class UserUpdate(APIView):
-    renderer_classes = [renderers.TemplateHTMLRenderer]
+@method_decorator(login_required, name="dispatch")
+class UserUpdate(UpdateView):
     template_name = "user_update.html"
+    form_class = SimpleUserChangeForm
+    admins_form_class = AdminUserChangeForm
+    model = CustomUser
 
-    @classmethod
-    def return_suitable_serializer_for_get_method(
-        cls, request: HttpRequest, user: CustomUser
-    ) -> Union[UserUpdateByAdminSerializer, UserUpdateSerializer]:
-        if request.user.user_type == CustomUser.UserType.ADMIN.name:
-            return UserUpdateByAdminSerializer(user, context={"request": request})
-        else:
-            return UserUpdateSerializer(user, context={"request": request})
-
-    @classmethod
-    def return_suitable_serializer_for_post_method(
-        cls, request: HttpRequest, user: CustomUser
-    ) -> Union[UserUpdateByAdminSerializer, UserUpdateSerializer]:
-        if request.user.user_type == CustomUser.UserType.ADMIN.name:
-            return UserUpdateByAdminSerializer(user, data=request.data, context={"request": request})
-        else:
-            return UserUpdateSerializer(user, data=request.data, context={"request": request})
+    def get_context_data(self, **kwargs: dict) -> dict:
```

Comment:
There is a method `get_form_class()` where you should put this logic, because it might be also used in other places (or ie. some 3rd party libs would call it).